music21/metadata/__init__.py

```diff
@@ -772,54 +728,28 @@ def all(self, skipContributors=False):
          ('filePath', '...corpus/corelli/opus3no1/1grave.xml'),
          ('localeOfComposition', 'Rome'),
          ('movementName', 'Sonata da Chiesa, No. I (opus 3, no. 1)'),
+         ('software', 'Dolet...'),
+         ('software', 'Finale...'),
          ('software', 'music21 v...'))
         '''
-        # pylint: disable=undefined-variable
-        allOut = {}
+        allOut: t.List[t.Tuple[str, str]] = []
 
-        for uniqueName in self.allUniqueNames:
-            try:
-                values: t.Tuple[str, ...] = self._getStringValuesByNamespaceName(
-                    properties.UNIQUE_NAME_TO_NAMESPACE_NAME[uniqueName]
-                )
-                if not values:
-                    continue
-                val = values[0]
-            except AttributeError:
-                continue
-            except KeyError:
-                # A uniqueName that doesn't have a PropertyDescription
-                # That's the three fileInfo properties, just get them
-                # attribute-style (i.e. as string)
-                if uniqueName == 'fileFormat':
-                    val = str(self.fileFormat)
-                elif uniqueName == 'filePath':
-                    val = str(self.filePath)
-                elif uniqueName == 'fileNumber':
-                    val = str(self.fileNumber)
-                else:
-                    raise
-
-            if skipContributors:
-                if self.isContributorUniqueName(uniqueName):
-                    continue
-            if val == 'None' or not val:
+        valueList: t.List[ValueType]
+        for name, valueList in self._contents.items():
+            if skipContributors and self._isContributorNamespaceName(name):
                 continue
-            allOut[uniqueName] = val
 
-        if not skipContributors:
-            for c in self.contributors:
-                if c.role in allOut:
-                    continue
-                if not c.name or c.name == 'None':
-                    continue
-                allOut[str(c.role)] = str(c.name)
+            uniqueName: t.Optional[str] = self.namespaceNameToUniqueName(name)
```

Comment:
why is the conversion to uniqueName happening on contents retrieval?  Wouldn't it be better to do it on contents set, so that retrieving routines don't need to do this extra step?

Reply:
Just pushed this exact change.